### PR TITLE
Add Salt Grid documentation

### PR DESCRIPTION
**Pull request by feliciackh**
uitk #381
---
[Grid]
Documentation

### CODE REVIEW

File: packages/grid/src/Grid.tsx [hunk 1]
```diff
@@ -99,7 +102,7 @@ export interface GridProps<T = any> {
   /**
    * Should return unique string for a given row data object.
    * If rowData is sparse then this function should work with undefined row data
-   * objects and create keys based on row index.
+   * objects and create keys based on row index. `(row: T, index: number) => string`
```
Comment (steveukx) [inline packages/grid/src/Grid.tsx:105]: As this is a defined type, does it need to be copied here. Worth checking whether the generated documentation will include the actual `RowKeyGetter<T> definition.

File: packages/grid/stories/grid.doc.stories.mdx [hunk 2]
```diff
@@ -17,27 +17,46 @@ import {
...
+npm install @jpmorganchase/uitk-grid;
+```
+
 Use the following call to import Grid.
 
 ```
 import { Grid } from "@jpmorganchase/uitk-grid";
 ```
 
+Note: ensure CSS `height` property for `.grid` has been set appropriately to see grid on screen
```
Comment (steveukx) [inline packages/grid/stories/grid.doc.stories.mdx:48]: Is the `.grid` a block element?

File: packages/grid/src/Grid.tsx (hunk 1 above)
Comment (feliciackh) [reply packages/grid/src/Grid.tsx:105]: @steveukx Grid props table in Grid docs is generated using Storybook <ArgsTable /> API. The way we use <ArgsTable />, Storybook infers data automatically from component and in our Grid component, we have our props `rowKeyGetter` typed as `RowKeyGetter<T>`; To make our props table generate the actual definition as I understand, will require us to update its type to `(row: T, index: number) => string` in Grid component. That's not as ideal hence why we've added it in the description instead 

ps. please let me know if anyone knows of a workaround

File: packages/grid/stories/grid.doc.stories.mdx [hunk 3]
```diff
@@ -66,43 +85,47 @@ You can easily create any project-specific column types using the same approach.
...
+
 <Canvas>
   <Story id="grid-new-grid--column-groups" decorators={[withFlexGap]} />
 </Canvas>
 
 ### Rows
 
-Grid rows are created by providing an array of objects in the `rowData` property of the grid.
+The grid rows are created by providing an array of objects in the `rowData` property of the grid.
 `rowData` also accepts sparse arrays.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:97]: Do you know what "sparse array" supposed to mean in this context? We might want to reword this, I have no idea what it meant.

File: packages/grid/stories/grid.doc.stories.mdx [hunk 4]
```diff
@@ -148,24 +169,60 @@ This grid also shows an example of a custom cell displaying a basic chart. Switc
 
 ### Editable cells
 
-Grid allows you to make cell values editable in place.
-Use the `CellEditor` component as a child of `GridColumn` to make the column editable.
-`CellEditor` is a non-rendered component.
-`CellEditor` is expected to have one child representing the specific editor component to use for the column, e.g. `TextCellEditor`, `DropdownCellEditor` or `NumericCellEditor`.
+The grid allows you to make cell values editable in place.
+
+Use the `CellEditor` (a non-rendered) component as a child of `GridColumn` to make the column editable.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:174]: Do we explain what a "non-rendered" component is anywhere in the docs? Might be useful to explain the concept, and it might prevent our users doing funky stuff with the children of the Grid in hopes of achieving different layouts.

File: packages/grid/stories/grid.doc.stories.mdx [hunk 5]
```diff
@@ -148,24 +169,60 @@ This grid also shows an example of a custom cell displaying a basic chart. Switc
...
+`CellEditor` is expected to have one child representing the specific editor component to use for the column.
 The child of the `CellEditor` component gets rendered in the focused cell when the grid goes to edit mode.
 
+We have [three sample components](https://uitk.pages.dev/?path=/story/grid-new-grid--editable-cells) created for use as `CellEditor`'s child,
+i.e. `TextCellEditor`, `DropdownCellEditor` and `NumericCellEditor`.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:179]: Let's remove DropdownCellEditor as it is not fully supported yet.

File: packages/grid/stories/grid.doc.stories.mdx [hunk 6]
```diff
@@ -148,24 +169,60 @@ This grid also shows an example of a custom cell displaying a basic chart. Switc
...
 
+To trigger edit mode on cells on your...
+
+- keyboard: `backspace` key _or_ `F2` key _or_ simply start typing
+- mouse: double click
+
 ### Pagination
 
-Implement pagination using the `Paginator` component. It is not a feature of the grid itself.
+[Implement pagination](https://uitk.pages.dev/?path=/story/lab-pagination--default-pagination) using the `Paginator` component. It is not a feature of the grid itself.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:193]: Maybe note that its support is experimental?

File: packages/grid/stories/grid.doc.stories.mdx [hunk 7]
```diff
@@ -148,24 +169,60 @@ This grid also shows an example of a custom cell displaying a basic chart. Switc
...
+```
 
 <Canvas>
   <Story id="grid-new-grid--grid-pagination" decorators={[withFlexGap]} />
 </Canvas>
 
+### Writing tests for Grid component
+
+The grid utilizes virtualization to load content and renders only what’s in the viewport.
+As previously fetched data leave the viewport, so do the DOM notes.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:221]:
```suggestion
As previously fetched data leave the viewport, so do the DOM nodes.
```

Comment (ZarkoPernar) [reply packages/grid/stories/grid.doc.stories.mdx:221]: Also, not sure I understand this sentence. What do you mean by "previously fetched data leave the viewport"?

File: packages/grid/stories/grid.doc.stories.mdx [hunk 8]
```diff
@@ -148,24 +169,60 @@ This grid also shows an example of a custom cell displaying a basic chart. Switc
...
+
+Note that if your test expects an element to be rendered, and it is out of the viewport, the test will not find the element - that is expected behaviour.
+To make tests work, ensure your grid is big enough and have scrolling implemented, closely mimicking how a user would experience using it.
```
Comment (ZarkoPernar) [inline packages/grid/stories/grid.doc.stories.mdx:224]: Also mention that virtual dom environments like JSDOM in jest don't have any dimensions and therefore won't render any rows.

**Review by sion-docs — DISMISSED** [review]
Made minor changes and approved

File: packages/grid/stories/grid.doc.stories.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx]: ahhh good catch - i had my samples' custom classNames set for `.grid` , will have it updated to` .uitkGrid `

File: packages/grid/stories/grid.doc.stories.mdx (hunk 3 above)
Comment (ZarkoPernar) [reply packages/grid/stories/grid.doc.stories.mdx:96]: Maybe rephrase it so it just says that the shape of the objects in the array can be anything?

File: packages/grid/stories/grid.doc.stories.mdx (hunk 7 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx:220]: Does "previously fetched data leave the viewport (data scrolled out of view in grid)" help provide better context?

File: packages/grid/src/Grid.css [hunk 9]
```diff
@@ -30,7 +30,7 @@
   --grid-separatorGap-default: 8px;
 }
 
-/* These ideally should be taken from characteristics */
+/* TODO update comment: These ideally should be taken from characteristics */
```
[the remark this answers is not in the document]
Comment (feliciackh) [reply packages/grid/src/Grid.css]: updated

File: packages/grid/src/Grid.css [hunk 10]
```diff
@@ -102,54 +103,150 @@
   --grid-row-height: var(--uitkGrid-row-height, var(--grid-row-height-default));
   --grid-padding: var(--uitkGrid-padding, var(--grid-padding-default));
   --grid-fontSize: var(--uitkGrid-fontSize, var(--grid-fontSize-default));
-  --grid-separatorGap: var(--uitkGrid-separatorGap, var(--grid-separatorGap-default));
-  --grid-header-fontSize: var(--uitkGrid-header-fontSize, var(--grid-header-fontSize-default));
-  --grid-groupHeader-fontSize: var(--uitkGrid-groupHeader-fontSize, var(--grid-header-fontSize));
-  --grid-groupHeader-fontWeight: var(--uitkGrid-groupHeader-fontWeight, var(--uitk-text-label-fontWeight-strong));
+  --grid-separatorGap: var(
```
[the remark this answers is not in the document]
Comment (feliciackh) [reply packages/grid/src/Grid.css:105]: updated

File: packages/grid/stories/grid.doc.stories.mdx (hunk 5 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx]: yup it's been removed

File: packages/grid/stories/grid.doc.stories.mdx (hunk 4 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx:174]: docs updated

File: packages/grid/stories/grid.doc.stories.mdx (hunk 6 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx:195]: docs updated

File: packages/grid/stories/grid.doc.stories.mdx (hunk 8 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx:229]: added a line to address Jest JSDOM, and recommend Cypress for testing

File: packages/grid/stories/grid.doc.stories.mdx (hunk 7 above)
Comment (feliciackh) [reply packages/grid/stories/grid.doc.stories.mdx]: gotcha, all done!

**Review by ararus — APPROVED** [review]

**Review by ararus — APPROVED** [review]
